### PR TITLE
Initial version of the API to get page content

### PR DESCRIPTION
**Pull request by OrKoN**
Hi,

please check this PR.

Questions:
- how should I structure the API? (I personally just need this one method)
- auth? rate-limiting? api-keys?
- how to properly mock pages service?
- do you separate request validators and response mappers?
- how do you document the API?

### CODE REVIEW

File: server/api/json.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ exports.register = function (server, options, next) {
 
   server.route({
     method: 'GET',
-    path: '/api/json/pages/{slug}/{revision}',
+    path: '/api/pages/{slug}/{revision}.json',
```
Comment (mathiasbynens) [inline server/api/json.js]: There should be no `/api/pages/` prefix. The URL should be `/{slug}/{revision}.json`. 

E.g.

* for `https://jsperf.com/slugs` the URL would be `https://jsperf.com/slugs.json`
* for `https://jsperf.com/slugs/3` the URL would be `https://jsperf.com/slugs/3.json`

This matches the `.atom` feeds for tests.

Sorry if my earlier comment wasn’t clear!

**Comment by OrKoN** [comment]
@mathiasbynens I have changed the path. Also I have removed the /api/json dummy along with its test. And I have added tests.

File: test/unit/server/api/json.js [hunk 2]
```diff
@@ -2,38 +2,91 @@ var Lab = require('lab');
...
+          published: '',
+          content: ''
+        }
+      ]
+    ];
+    pagesServiceStub.getBySlug = function (cnt) {
+      return Promise.resolve(data);
+    };
     server.inject(request, function (response) {
-      Code.expect(response.statusCode).to.equal(200);
```
Comment (maxbeatty) [inline test/unit/server/api/json.js]: can we keep the status code and header assertions?

Comment (OrKoN) [inline test/unit/server/api/json.js]: I have added these checks too. Also rebased and squashed commits

**Comment by OrKoN** [comment]
Hey, are there any TODOs left in order to merge this?

**Comment by OrKoN** [comment]
@maxbeatty is it possible to disable cloudfare for API routes? for example, `curl -v -H "Accept: application/json" -X GET https://jsperf.com/number-to-string/58.json` returns a cloudfare ddos protection page.

**Pull request merged**